### PR TITLE
refactor(build/efb): aircraft project prefix overhaul to better support different a/c and variants (v0.1)

### LINKED ISSUE #8499: Build_Info.json not loaded within EFB because of non-reliable Aircraft Check
### Aircraft Version

Development

### Build info

```json
{
    "built": "2024-02-14T18:56:54+00:00",
    "ref": "refs/heads/master",
    "sha": "8501776e42dda002ce1270d9516adcdfe14bd75e",
    "actor": "aguther",
    "event_name": "manual",
    "pretty_release_name": "master:8501776e",
    "version": "a32nx-v0.12.0-dev.8501776"
}
```

### Describe the bug

The introduction of PR #8472 added a new type check for aircraft, focusing on verifying the titles of the aircraft against a list of pre-defined names. This measure was implemented to ensure consistency and accuracy in aircraft identification. However, a potential issue arises from the variability in how livery authors might title their aircraft liveries. Due to inconsistencies in naming conventions used by livery authors, there's a risk that this type check might not recognize some aircraft titles as expected. When the type check fails to identify an aircraft title correctly, it will result in the build_info.json information not being loaded. This could lead to complications, highlighting the need for a more flexible or comprehensive approach to aircraft verification.

grafik

Example of a bad livery aircraft.cfg

```
[VERSION]
major = 1
minor = 0

[VARIATION]
base_container = "..\FlyByWire_A320_NEO"

;===================== FLTSIM =====================

[FLTSIM.0]
title = "A32NX Lufthansa D-AIJC" ; Variation name
model = "" ; model folder
panel = "" ; panel folder
sound = "" ; sound folder
```

The [fbw-common/src/systems/shared/src/aircraftTypeCheck.ts](https://github.com/flybywiresim/aircraft/pull/8472/files#diff-ad50c9e88bed08fb13dd1a0182a35f6bbb907c66e335ddfb7fa2c7a285447952) will search up the SimVar Title for A320 but only will find A32NX as case "other" and not loading up the build_info.json.

### Expected behavior

This is exactly the expected behavior, another approach is needed to be more consistent.

### Steps to reproduce

1. Load the plane with a livery which has a "bad" title: e.g. `title = "A32NX Lufthansa D-AIJC" ; Variation name`
2. Open up EFB and switch to About page in Settings.

### References (optional)

_No response_

### Additional info (optional)

Another approach could be to add a new SimVar called "A32NX_AIRCRAFT_TYPE" and to check against the value of this SimVar. Still possible to change via the Debug Tools on runtime, but this could be only a edge case.

### Discord Username (optional)

Revyn112

### PR DESCRIPTION
**Pull request by 2hwk**
Partially Addresses #7665
Fixes #8499

## Summary of Changes

Based on #8500 we will start to transition away from a panel.cfg/L:Var based approach towards a markup .json and build time  based approach to handling multiple A/C, variants as well as liveries (by ATC tail registration)

This will ensure that the aircraft prefix is correctly set (not dependant on livery), as proposed by #8500, as well as allowing for easier flyPad configuration, management and changes in the future. 

This modifies:

flyPad OS (3.2.1)
- [x] Ground > Payload
  - [x] Use VFS config loaded from redux store 
  - [x] Optimised logic slightly
  - [x] Ported over 32 changes to 38
  - [x] Save deck selection in redux store
- [x] Ground > Services
  - [x] Use VFS config loaded from redux store 
- [x] Ground > Fuel
  - [x] Use VFS config loaded from redux store 
- [x] Ground > Pushback
  - [x] Use VFS config loaded from redux store 
- [x] Checklists
   - [x]  Use VFS config loaded from redux store
- [x] Dispatch > Overview 
  - [x] Use VFS config loaded from redux store 
- [x] Settings > Throttle Config
  - [x] Use VFS config loaded from redux store 
- [x] Settings > About
  - [x] Add SimVar "Title" to about page debug
  
- [x] Clean up fbw-aXX > fbw-common EFB refactor

Potential:
- [x] ~~FBW Api - Add "title" and "aircraft" to FBW api request~~ Not done

Extras-host
 - [x] Added new module to load configuration from VFS/{aircraft}/{variant} json5 files
    - [x] ~~Override with VFS/{aircraft}/{variant}/{atc_id} json5~~ (not in this PR)

This also adds a new ``AIRCRAFT_PROJECT_PREFIX`` environ to the project ``.env`` which is mandatory. 

_Note: ``AIRCRAFT_VARIANT``  has been added, but in the future this will be created with the script/*.sh scripts._

## Additional context

Discord username (if different from GitHub): 2cas

## Testing instructions

- Test all the following EFB pages, ensure functionality is as expected. 

  - FlyPad Widget/Dashboard
     - Visual: Correct Aircraft Variant is displayed
  - Ground > Payload
    - Visual: Correct page is displayed - Graphics, text, etc.
    - Fuction: GSX Payload
    - Function: Non GSX manual loading
    - Function: Simbrief Loading
  - Ground > Services
    - Visual: Correct page is displayed - Graphics, text, etc.
    - Function: Basic check
  - Ground > Fuel
    - Visual:  Correct page is displayed  - Graphics, text, etc.
    - Function: Basic check
  - Ground > Pushback
    - Visual: Correct page is displayed  - Graphics, text, etc.
    - Function: Basic check
 - Checklists
    - Visual: Correct page is displayed -  Graphics, text, etc.
    - Function: Basic check
  - Dispatch > Overview 
    - Visual: Correct page is displayed  - Graphics, text, etc.
    - Visual: Aircraft information is correct for type
  - Settings > Throttle Config
    - Functionality check - Throttle configuration loads
    - Functionality check - remove ThrottleConfig.ini - ensure default values work
    - Functionality check - create new ThrottleConfig.ini via using the axis UI.
  - Settings > About 
    - New livery title field in about page 
  - General
    - EFB works when powered on initially and navigating to all the above pages
    - EFB works when power cycled off then on and navigating to all the above pages
    - EFB works when battery power is depleted and navigating to all the above pages
    - **EFB works with various liveries** i.e. https://flightsim.to/file/22878/fbw-a32nx-lufthansa-livery-d-aija
    - EFB checklists are correctly cleared when the EFB is power cycled off/on
    - EFB works when battery is depleted (use ingame time to manipulate and ensure battery simulation is on in settings)

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

**Comment by 2hwk** [comment]
Testing List:

- [ ] Test all the following EFB pages, ensure functionality is as expected. 

  - [ ] Ground > Payload
    - [ ] Visual: Correct page is displayed - Graphics, text, etc.
    - [ ] Fuction: GSX Payload
    - [ ] Function: Non GSX manual loading
    - [ ] Function: Simbrief Loading
  - [ ] Ground > Services
    - [ ] Visual: Correct page is displayed - Graphics, text, etc.
    - [ ] Function: Basic check
  - [ ] Ground > Fuel
    - [ ] Visual:  Correct page is displayed  - Graphics, text, etc.
    - [ ] Function: Basic check
  - [ ] Ground > Pushback
    - [ ] Visual: Correct page is displayed  - Graphics, text, etc.
    - [ ] Function: Basic check
  - [ ] Dispatch > Overview 
    - [ ] Visual: Correct page is displayed  - Graphics, text, etc.
    - [ ] Visual: Aircraft information is correct for type
  - [ ] Settings > Throttle Config
    - [ ] Functionality check - Throttle configuration loads
    - [ ] Functionality check - remove ThrottleConfig.ini - ensure default values work
    - [ ] Functionality check - create new ThrottleConfig.ini via using the axis UI.
  - [ ] General
    - [ ] EFB works when powered on initially and navigating to all the above pages
    - [ ] EFB works when power cycled off then on and navigating to all the above pages
    - [ ] EFB works when battery power is depleted and navigating to all the above pages

- [ ] Open ModelBehaviors > LocalVariables debug window. A32NX_AIRCRAFT_TYPE should be set correctly according to simvar.md i.e. 

      ```
      - A32NX_AIRCRAFT_TYPE
          - Enum
          - Indicates the aircraft type
              Aircraft | Value
              --- | ---
              Unknown | 0
              A320_251N | 1
              A380_842 | 2
      ```

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/config/a32nx/a320-251n/README.md
Comment (frankkopp) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/config/a32nx/a320-251n/README.md:9]: Let's add json5 and make it standard - for config files json5 makes more sense as these should often be commented.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/config/a32nx/a320-251n/README.md
Comment (frankkopp) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/config/a32nx/a320-251n/README.md:16]: .json5

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/config/a32nx/a320-251n/README.md
Comment (frankkopp) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/config/a32nx/a320-251n/README.md:22]:
```
Please consult with our core development team before adding more configuration and hardcoded values into the ``config.json5`` markup.
```

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/config/a380x/a380-842/README.md
Comment (frankkopp) [inline fbw-a380x/src/base/flybywire-aircraft-a380-842/config/a380x/a380-842/README.md:11]: .json5

**Review by frankkopp — APPROVED** [review]
LGTM - a bit more documentation would have been nice ;)

File: fbw-common/src/systems/shared/src/UniversalConfigProvider.ts
Comment (2hwk) [reply fbw-common/src/systems/shared/src/UniversalConfigProvider.ts:1]: Maybe in a future PR, but there's quite a bit to process later down the line anyway which may require a different boilerplate entirely. I don't agree that there is a need to split at this point, there is no verification at the moment beyond reading from json5.

**Comment by 2hwk** [comment]
> LGTM - a bit more documentation would have been nice ;)

Yes, earmarked for v0.2 when I've gotten a better handle on the functional requirements and what boilerplate, middleware, etc is needed, as well as that commbus link with WASM.

**Comment by noonebling** [comment]
Quality Assurance Trainee Report

Discord          : noonebling
Object of testing: 8599
Tier of Testing  : 1
Date             : 07/04/2024

Testing Process: 
Tested all the EFB pages, with thorough checks on grahics, text, etc on Ground Payload, Services, Fuel and Pushback. Done with GSX and Simbrief integrated. 
Tested Checklist pages for graphics, text and functions.
Checked Dispatch pages for graphics, text, aircraft information
Did functionality checks for throttle configs
New livery title field in about page works for different liveries
Checked every feature in the General testing instructions

Negatives:
N/A

Testing Results: 
Passed

Conclusions: 
Works as expected

**Comment by 2hwk** [comment]
As this could create merge conflicts with fms-v2, I'll hold off on merging until that is merged.

**Comment by 2hwk** [comment]
QA pass label removed to let more QA trainees be able to run through this one while we wait for some large PR merges.

**Pull request merged**
